src/objective-c/GRPCClient/private/GRPCCore/GRPCCompletionQueue.mm

```diff
@@ -69,6 +69,9 @@ - (instancetype)init {
             case GRPC_QUEUE_SHUTDOWN:
               grpc_completion_queue_destroy(unmanagedQueue);
               return;
+            case GRPC_QUEUE_TIMEOUT:
+              NSLog(@"GRPC_QUEUE_TIMEOUT, success: %d, tag: %p", event.success, event.tag);
+              break;
```

Comment:
Would this change silently disregard rpc errors?

I wonder if we should call `handler` with a failure event?

Reply:
Not very sure but my understanding is that the completion queue is not supposed to timeout unless passed a finite timeout value.
The change in this PR will retry getting a event from the completion queue, which may indicate connection timeout in the event.

The crash report only had the stack trace, it may be helpful if we have the logs before the crash.

Reply:
I tried return an error from [pollset_work](https://github.com/grpc/grpc/blob/master/src/core/lib/iomgr/ev_apple.cc#L263) to simulate the queue timeout error, it appears the handler was not setup and trying to use it to propagate the error still crashes.


<img width="942" alt="Screenshot 2024-04-21 at 8 44 07 AM" src="https://github.com/grpc/grpc/assets/28025951/c2e67b67-744d-42dd-80fb-4b78f9cb2849">

Reply:
Aha ok thank you! So are you saying that you'd prefer to eliminate the crash in this PR as well? (Besides only adds logging?)

In any case I trust your judgement here since you've given enough thoughts on this 😃 

We can chat when we meet :)

Reply:
Please let me know what the client should expect to find after this change. (e.g. the crash goes away? And when to check logs? etc.) 😃